variables.tf

```
@@ -136,3 +136,15 @@ variable "root_block_device_encrypted" {
   default     = true
   description = "Whether to encrypt the root block device"
 }
+
+variable "create_public_subnets" {
+  type        = bool
+  default     = null
```

Because of a bug in the Terraform registry (hashicorp/terraform#21417), the registry shows inputs with default values of `null` as  required when in fact they are optional. Also, there is no practical advantage of using a default value of `null` here, we want the default to be `true`. 

```suggestion
  default     = true
```